packages/ui/src/components/DataMapper/on-paste-data-mapper.ts

```diff
@@ -0,0 +1,119 @@
+import { IVisualizationNode } from '../../models';
```

Comment:
I think we can remove the need for the recursion here in favor of using
https://github.com/KaotoIO/kaoto/blob/4b97599ae4f0015d297e115265665fd9080e0657/packages/ui/src/utils/get-viznodes-from-graph.ts#L22-L37

This function iterates over the `IVisualizationNode` tree recursively and finds elements based on a predicate, so you could receive the `vizNode` in these functions, pick the `vizNodes` based on the `isDataMapperNode` filter, and then operate on a single array.

Reply:
We can find the `GraphElement` in the `useController` hook. There's a caveat, though: it will search within the visible nodes.

You're right, we don't have the `vizNode` available from the clipboard, so let's keep it like this and revisit once we complete the clipboard parsing. This way, we could have stale `vizNodes` from the clipboard.